doc/Architecture.md

```diff
@@ -279,7 +279,81 @@ The offchain code that constructs the DeNS database will always treat the _most
 
 ## Architecture - Arweave (Technical) (WIP)
 
-TODO
+It seems that the maximum amount of data we can upload paying the minimum amount possible is:
+
+```bash
+https://arweave.net/price/256000
+212017846
+```
+
+This price is given in Wilson, where a Wilson is related to Ar (the arweave currency) as:
+
+```python
+Ar = 1e12 wilson
+```
+
+At today the price of an Ar is around `9.5` usd, this means that the minimum cost of a transaction is : `0.002014169537` usd.
+And the maximum size of a transaction with such a price is above 250 KB.
+
+Additionally some gateways allows the upload of certain amount of bytes for free.
+
+Those are the reason we choose to store every set of records in a particular zone file as a single transaction.
+
+### Initial Upload
+
+Initially we have a setup like this:
```

Comment:
```suggestion
Initially, we will assume we have the following zone files.
```
Iirc, commas should follow the adverb at the beginning of a sentence?

Maybe it would be good to recall to the reader _what_ a zone file is?